### ISSUE TITLE
Exception Value: 'str' object has no attribute 'name'

### ISSUE THREAD

**Issue by fpoulain** [makinacorpus/django-tracking-fields#6]
Got this exception after upgrading my project : 'str' object has no attribute 'name'

Django Version: 2.2.8
Python Executable: /usr/bin/uwsgi-core
Python Version: 3.7.3

Traceback:
```
File "venv/lib/python3.7/site-packages/django/core/handlers/exception.py" in inner
  34.             response = get_response(request)

File "venv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  115.                 response = self.process_exception_by_middleware(e, request)

File "venv/lib/python3.7/site-packages/django/core/handlers/base.py" in _get_response
  113.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "venv/lib/python3.7/site-packages/django/views/generic/base.py" in view
  71.             return self.dispatch(request, *args, **kwargs)

File "venv/lib/python3.7/site-packages/django/utils/decorators.py" in _wrapper
  45.         return bound_method(*args, **kwargs)

File "venv/lib/python3.7/site-packages/django/contrib/auth/decorators.py" in _wrapped_view
  21.                 return view_func(request, *args, **kwargs)

File "./pgcd/pgcd/mixins.py" in dispatch
  63.         return super(LoginRequired, self).dispatch(*args, **kwargs)

File "venv/lib/python3.7/site-packages/django/views/generic/base.py" in dispatch
  97.         return handler(request, *args, **kwargs)

File "venv/lib/python3.7/site-packages/django/views/generic/edit.py" in post
  194.         return super().post(request, *args, **kwargs)

File "venv/lib/python3.7/site-packages/django/views/generic/edit.py" in post
  142.             return self.form_valid(form)

File "venv/lib/python3.7/site-packages/django/views/generic/edit.py" in form_valid
  125.         self.object = form.save()

File "venv/lib/python3.7/site-packages/django/forms/models.py" in save
  458.             self.instance.save()

File "venv/lib/python3.7/site-packages/django/db/models/base.py" in save
  741.                        force_update=force_update, update_fields=update_fields)

File "venv/lib/python3.7/site-packages/django/db/models/base.py" in save_base
  790.                 update_fields=update_fields, raw=raw, using=using,

File "venv/lib/python3.7/site-packages/django/dispatch/dispatcher.py" in send
  175.             for receiver in self._live_receivers(sender)

File "venv/lib/python3.7/site-packages/django/dispatch/dispatcher.py" in <listcomp>
  175.             for receiver in self._live_receivers(sender)

File "venv/lib/python3.7/site-packages/tracking_fields/tracking.py" in tracking_save
  351.             _create_update_tracking_event(instance)

File "venv/lib/python3.7/site-packages/tracking_fields/tracking.py" in _create_update_tracking_event
  205.                     _create_tracked_field(event, instance, field)

File "venv/lib/python3.7/site-packages/tracking_fields/tracking.py" in _create_tracked_field
  176.         old_value=_serialize_field(old_value),

File "venv/lib/python3.7/site-packages/tracking_fields/tracking.py" in _serialize_field
  144.         return json.dumps(field.name,
```

I munged it with
```diff
--- venv/lib/python3.7/site-packages/tracking_fields/tracking.py	2019-12-10 19:16:24.032967867 +0100
+++ venv/lib/python3.7/site-packages/tracking_fields/tracking.py.old	2019-12-10 19:16:00.972746619 +0100
@@ -140,7 +140,7 @@
     if isinstance(field, Model):
         return json.dumps(six.text_type(field),
                           ensure_ascii=False)
-    if isinstance(field, StateWrapper) and not isinstance(field, str):
+    if isinstance(field, StateWrapper):
         return json.dumps(field.name,
                           ensure_ascii=False)
     try:
```

**Comment by Gagaro** [comment]
What is the field you are tracking which raise this exception?

**Comment by youri-ds** [comment]
Hello
I have got the same issue with Django 2.2.8 and any field.
The patch is working but is it correct? 
Because I don't understand this code: tracking.py (174)
```
    return TrackedFieldModification.objects.create(
        event=event,
        field=fieldname,
        old_value=_serialize_field(old_value),
        new_value=_serialize_field(getattr(instance, field))

```
The field becomes a string.

**Comment by Gagaro** [comment]
It's fixed in 1.2.0 with the Django 3.0 compatibility.

I had to drop 1.11 compatibility too, though it should only break with python 2.

**Comment by fpoulain** [comment]
thanks!
